cppcache/src/PoolFactory.cpp

```diff
@@ -271,24 +271,26 @@ std::shared_ptr<Pool> PoolFactory::create(const char* name) {
       if (copyAttrs
               ->getThreadLocalConnectionSetting() /*&& !copyAttrs->getPRSingleHopEnabled()*/) {
         // TODO: what should we do for sticky connections
-        poolDM =
-            std::make_shared<ThinClientPoolStickyDM>(name, copyAttrs, tccm);
+        poolDM = std::make_shared<ThinClientPoolStickyDM>(name.c_str(),
+                                                          copyAttrs, tccm);
       } else {
         LOGDEBUG("ThinClientPoolDM created ");
-        poolDM = std::make_shared<ThinClientPoolDM>(name, copyAttrs, tccm);
+        poolDM =
+            std::make_shared<ThinClientPoolDM>(name.c_str(), copyAttrs, tccm);
       }
     } else {
       LOGDEBUG("ThinClientPoolHADM created ");
       if (copyAttrs
               ->getThreadLocalConnectionSetting() /*&& !copyAttrs->getPRSingleHopEnabled()*/) {
-        poolDM =
-            std::make_shared<ThinClientPoolStickyHADM>(name, copyAttrs, tccm);
+        poolDM = std::make_shared<ThinClientPoolStickyHADM>(name.c_str(),
+                                                            copyAttrs, tccm);
       } else {
-        poolDM = std::make_shared<ThinClientPoolHADM>(name, copyAttrs, tccm);
+        poolDM =
+            std::make_shared<ThinClientPoolHADM>(name.c_str(), copyAttrs, tccm);
       }
     }
 
-    cacheImpl->getPoolManager().addPool(name,
+    cacheImpl->getPoolManager().addPool(std::move(name),
```

Comment:
Why `move`?

Reply:
Why not? The method is `PoolFactory::create(std::string name)` so I am moving the already copied value into the member variable, rather than another copy. Saves the copy of the underlying `char[]`.

Reply:
I feel like it makes it just a little easier to shoot ourselves in the foot (e.g. accidentally editing to use name after this line).  Also since `PoolFactory::create` probably won't be called in a tight loop a copy doesn't seem too terrible?